core/src/main/scala/org/apache/spark/scheduler/SchedulableBuilder.scala

```diff
@@ -69,19 +72,29 @@ private[spark] class FairSchedulableBuilder(val rootPool: Pool, conf: SparkConf)
   val DEFAULT_WEIGHT = 1
 
   override def buildPools() {
-    var is: Option[InputStream] = None
+    var fileData: Option[FileData] = None
     try {
-      is = Option {
-        schedulerAllocFile.map { f =>
-          new FileInputStream(f)
-        }.getOrElse {
-          Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
+      fileData = schedulerAllocFile.map { f =>
+        Some(FileData(new FileInputStream(f), f))
+      }.getOrElse {
+        val is = Utils.getSparkClassLoader.getResourceAsStream(DEFAULT_SCHEDULER_FILE)
+        if(is != null) Some(FileData(is, DEFAULT_SCHEDULER_FILE))
+        else {
+          logWarning(s"No Fair Scheduler file found.")
+          None
         }
       }
 
-      is.foreach { i => buildFairSchedulerPool(i) }
+      fileData.foreach { data =>
+        logInfo(s"Fair Scheduler file: ${data.fileName} is found successfully and will be parsed.")
```

Comment:
nit: I find this a little confusing in the case where the default filename was used -- since the user didn't actually specify that file.  Can you log separately in the two cases?   So to use Mark's suggested message, s"Creating Fair Scheduler pools from ${data.fileName}" in the first case, and in the second case, s"Creating Fair Scheduler pools from default file ($DEFAULT_SCHEDULER_FILE). That way you can also keep the old code organization, which was a bit clearer.